### ISSUE TITLE
Regeneration of coarse-grain results issue.

### ISSUE THREAD

**Issue by MFarhatUllah** [YuanJiangGit/StagedVulBERT#2]
“Hi, we used model_2048.bin from checkpoint-best-f1 and followed all instructions. But the results are near-zero F1. Could you please re-share the correct checkpoint used in the paper?” or your model has some issue ? 

I have downloaded pretrained model, fine-tuned model, test.csv, val.csv, codebert from microsoft complete. And placed as per your directions. But failed to produce F1-score = 92. 

Could you please guide me?

Here is the terminal query:
(V2GPUpython3.10) aaa@farhat:~/Documents/StagedVulBERT-master$ python Entry/StagedBert_vul.py   --do_test   --model_name_or_path resource/codebert-base   --tokenizer_name resource/codebert-base   --initialize_model_name resource/output/checkpoint-best-f1/model_2048.bin   --test_data_file resource/dataset/test.csv   --output_dir resource/output
2025-06-05 19:12:35.621134: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-06-05 19:12:35.628064: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1749121955.636974   21260 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1749121955.639646   21260 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-06-05 19:12:35.648728: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
/home/aaa/Documents/StagedVulBERT-master/Entry/StagedBert_vul.py:536: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model.load_state_dict(torch.load(args.initialize_model_name, map_location=args.device), strict=False)
✅ Loaded weights from resource/output/checkpoint-best-f1/model_2048.bin
100%|█████████████████████████████████████████████████████████████████████████████| 18864/18864 [00:04<00:00, 4484.36it/s]
parse done!
saved at %s resource/dataset/test_lp.pkl
100%|███████████████████████████████████████████████████████████████████████████████████| 590/590 [04:58<00:00,  1.97it/s]
y_trues: [0 0 0 ... 0 0 0]
best_threshold:  0.5
y_preds: [False False False ... False False False]
✅ Saved evaluation predictions to: resource/dataset/test_eval_predictions.csv
#############################################
#################################################

here is the log file generated by your code.
06/05/2025 19:12:36 - WARNING - __main__ -   device: cuda, n_gpu: 1
06/05/2025 19:12:38 - INFO - __main__ -   Training/evaluation parameters Namespace(train_data_file='../resource/dataset/train.csv', output_dir='resource/output', block_size=512, eval_data_file='../resource/dataset/val.csv', test_data_file='resource/dataset/test.csv', model_name='model_2048.bin', model_name_or_path='resource/codebert-base', config_name='../resource/codebert-base', use_non_pretrained_model=False, tokenizer_name='resource/codebert-base', seg_num=4, do_train=False, do_test=True, evaluate_during_training=False, train_batch_size=32, eval_batch_size=32, gradient_accumulation_steps=1, learning_rate=2e-05, weight_decay=0, adam_epsilon=1e-08, max_grad_norm=1.0, max_steps=-1, warmup_steps=0, seed=12345, epochs=10, num_attention_heads=12, initialize_model_name='resource/output/checkpoint-best-f1/model_2048.bin', n_gpu=1, device=device(type='cuda'))
06/05/2025 19:12:47 - INFO - __main__ -     Num examples = 18864
06/05/2025 19:12:47 - INFO - __main__ -     Batch size = 32
06/05/2025 19:17:46 - INFO - __main__ -   ***** Eval results *****
06/05/2025 19:17:46 - INFO - __main__ -     eval_IoU = 0.0009
06/05/2025 19:17:46 - INFO - __main__ -     eval_accuracy = 0.9441
06/05/2025 19:17:46 - INFO - __main__ -     eval_f1 = 0.0019
06/05/2025 19:17:46 - INFO - __main__ -     eval_precision = 1.0
06/05/2025 19:17:46 - INFO - __main__ -     eval_recall = 0.0009
06/05/2025 19:17:46 - INFO - __main__ -     eval_threshold = 0.5

**Comment by YuanJiangGit** [comment]
It is very likely that the issue is caused by your local environment settings. So far, we haven’t encountered similar reproducibility problems reported by other users. I would suggest trying to debug the code using PyCharm, which may help you better identify the issue, rather than running it directly from the command line.
